test/code/plugins/oms_common_test.rb

```diff
@@ -209,6 +357,38 @@ def test_get_hostname_in_containerized_agent
       assert_equal(test_hostname, hostname, 'get_hostname should read from Socket.gethostname')
     end
 
+    def test_get_hostname_mocked
```

Comment:
You're mocking in `test_get_hostname_as_ipv4_address` as well, so the test name "test_get_hostname_mocked" is not precise. "test_get_hostname_as_alpha_string" might be better